### PR TITLE
Focus and mouse events handlers enhanced. Closes #517

### LINKED ISSUE #517: Can't gain focus of ComboBox with mouse right after initialization
I can repro on both Windows and Linux.

1. Run `UI Catalog`
2. Start `ListView & ComboBox` or scenario
3. Try to mouse click on ComboBox's edit

Expected:
The ComboBox gets focus.

Actual:
Nothing happens.

4. Press TAB

Expected:
The ComboBox gets focus

Actual:
The ComboBox gets focus

5. Mouse click on List.
6. Mouse click on ComboBox

Expected:
The ComboBox gets focus

Actual:
The ComboBox gets focus

### PR DESCRIPTION
**Pull request by BDisp**

### CODE REVIEW

File: Terminal.Gui/Core.cs [hunk 1]
```diff
@@ -1461,24 +1494,67 @@ public override string ToString ()
...
+		public class MouseEventEventArgs : EventArgs {
+			/// <summary>
+			/// Constructs.
+			/// </summary>
+			/// <param name="me"></param>
+			public MouseEventEventArgs (MouseEvent me) => mouseEvent = me;
+			/// <summary>
+			/// The <see cref="MouseEvent"/> for the event.
+			/// </summary>
+			public MouseEvent mouseEvent { get; set; }
```
Comment (tig) [inline Terminal.Gui/Core.cs:1509]: Public property names should be `CamelCase`. `mouseEvent` should be `MouseEvent`.

**Comment by BDisp** [comment]
Ups my bad sorry. The ideal was I made the change to the `Responder `class and create there the method `OnMouseEvent `but that would brokens the `MouseEvent `method. What do you want meaning about I label the PR with `breaking-change`. How do I do that, sorry but I never did.

**Comment by BDisp** [comment]
I don't see any option to put a label but only to filter. Then you can guide me in this, please.

File: Terminal.Gui/Core.cs (hunk 1 above)
Comment (BDisp) [reply Terminal.Gui/Core.cs]: I don't see any option to put a label but only to filter. Then you can guide me in this, please.

**Comment by tig** [comment]
> Ups my bad sorry. The ideal was I made the change to the `Responder `class and create there the method `OnMouseEvent `but that would brokens the `MouseEvent `method. What do you want meaning about I label the PR with `breaking-change`. How do I do that, sorry but I never did.

I just did it:

image

**Comment by tig** [comment]
I'm almost done with the code reorg. It will be easier for you to rebase this commit against that, than have me incorporate this change. "core.cs" no longer exists. So please hold on this (and other PRs) until I accept that PR. Ok?

**Comment by tig** [comment]
Please resubmit. Note `core.cs` no longer exists and you'll have to merge these changes into the right place.

**Comment by BDisp** [comment]
I will do that tomorrow.

**Comment by tig** [comment]
Well done.

**Pull request merged**